endive-playground/src/scripts/generators/latex.js

```diff
@@ -0,0 +1,116 @@
+
+export const latexGenerator = new Blockly.Generator("Endive");
+
+latexGenerator.RESERVED_WORDS_ = "let,lemma,example,ohno";
+latexGenerator.init = function (workspace) {
+  this.nameDB_ = new Blockly.Names(latexGenerator.RESERVED_WORDS_);
+  latexGenerator.nameDB_.setVariableMap(workspace.getVariableMap());
+};
+
+/*latexGenerator.scrub_ = function (block, code, thisOnly) {
+  const nextBlock = block.nextConnection && block.nextConnection.targetBlock();
+  if (nextBlock && !thisOnly) {
+    return code + latexGenerator.blockToCode(nextBlock);
+  }
+  return code;
+};*/
+
+latexGenerator["theorem"] = function (block, generator) {
+  const statement = generator.valueToCode(block, "STATEMENT", 0);
+
+  var proof = "";
+  var proofBlock = block.getInput("PROOF").connection.targetBlock();
+
+  while (proofBlock) {
+    var tactic =
+      generator.blockToCode(proofBlock).replace(/^/gm, "  ") +
+      "\n";
+    proof += tactic;
+    proofBlock = proofBlock.getNextBlock();
+  }
+  if (statement === "") {
+    return ""
+  }
+  const code = `Théorème : $${statement}$<br/> Proof : <br/>${proof} <br/>`;
+  return code;
+};
+
+latexGenerator["lemma"] = function (block, generator) {
+  const name = block.getFieldValue("NAME");
+  const statement = generator.valueToCode(block, "STATEMENT", 0);
+
+  var proof = "";
+  var proofBlock = block.getInput("PROOF").connection.targetBlock();
+
+  while (proofBlock) {
+    var tactic =
+      generator.blockToCode(proofBlock).replace(/^/gm, "  ") +
```

Comment:
Does this work? I think browsers collapse spaces and ignore line breaks in HTML.